docs/developer/architecture/user_customisations.rst

```diff
@@ -0,0 +1,51 @@
+User Customisations and Persistence
+===================================
+
+Context
+-------
+
+MetalK8s comes with a set of tools and services that may need to be configured
+on site. At the same time, we don't want the administrator of the cluster to
+master each and every service of the cluster. We also don't want to allow any
+kind of customisations since it would make the system even more complex to test
+and maintain over the time.
+
+In addition to those services, MetalK8s deployment may have to be adapted
+depending on the architecture of the platform or depending on the different
+use cases running on the top of it.
+
+Be it services or MetalK8s customisations, we need to ensure it is persisted and
+resistant to various type of events such as node reboot, upgrade, downgrade,
+backup, restore
+
+.. _usercusto-requirements:
+
+Requirements
+------------
+
+As a MetalK8s administrator, I have access to a finite list of settings I can
+customise on-site in order to match with my environment specificities:
+
+- list of static users and credentials configured in Dex
+- Integration with an external IDP configuration in Dex
+- Alert notification configuration in Alert Manager
+- Number of replicas for the Prometheus and Alert Manager cluster
+- Other items will appear as we add new customisable features in MetalK8s
+
+Depending on the the setting, impact of the customisation can be at cluster
+level or namespace level (not sure it is needed)
```

Comment:
None of the use cases above require cluster-level config, but if you have ideas where this would be required, we'll need to take it into account for the design (which uses namespace-scoped ConfigMaps at the moment).

Reply:
Dex configuration is not a cluster-level config ?

Reply:
Everything is cluster-level config, in a sense that we won't have different configurations of the same service in different places. But configuration will always be tied to the service itself, which sits in a namespace managed by MetalK8s.
Also, this makes the name of this feature really misleading: there's nothing related to "users" here, except that it's a user who will do the configuration.

Reply:
would you agree with MetalK8s customizations or cluster customizations?

Reply:
Went with `Cluster and Services Configurations and Persistence`.